ledger-core/network/watermill/router.go

```diff
@@ -21,41 +21,41 @@ import (
 	"github.com/insolar/assured-ledger/ledger-core/vanilla/throw"
 )
 
-var _ messagesender.MessageRouter = watermillRouter{}
+var _ messagesender.MessageRouter = Router{}
 
-func newWatermillRouter(ctx context.Context, outHandler message.NoPublishHandlerFunc) watermillRouter {
+func NewRouter(ctx context.Context, outHandler message.NoPublishHandlerFunc) Router {
```

Comment:
Can we make separate function for `.Pub.Publish` and setter for "alternative" publisher?

Reply:
Why do we need it?

Reply:
Something like `Router.Publish` and `Router.ReplacePublisher`/`Router.SetPublisher`.

Reply:
I'm not sure that router should publish. And I'm not modifing publisher. We are using the same strategy when sending to self.

Reply:
Router implements `Publisher` interface - it's normal situation, while `router` has 1 publisher and 1 subscriber. And it'll be the same way, since we have "topics" in our "publisher"

Reply:
Besides our "router" isn't real router. It's transparent wrapper for 1 wm publisher and 1 wm subscriber :)

Reply:
OK, I can create Publish and revert pub to package scope

Reply:
Added PublishMesssage and ReplacePublisher. Review please